### ISSUE TITLE
Deal with lava while tunneling or digging a hole

### ISSUE THREAD

**Issue by SergeiBond** [cabaletta/baritone#1794]
My main use of B. is to automate things, especially tedious construction projects like tunneling subways or digging holes to house various redstone machines.

The major problem is that when you create a selection for a future tunnel, it just digs through the blocks that are adjacent to lava, releases the lava and gets damaged by it.

I suggest, at the very least, an option to avoid releasing lava, or--even better--an option to erase lava blocks that are inside or are adjacent to the selected space by filling them out with throwaway materials.

**Comment by dominikaaaa** [comment]
Duplicate of #1560

**Comment by SergeiBond** [comment]
I don't agree that this is a duplicate of that suggestion.
That person is talking about pathfinding across areas with flowing lava, especially in freshly-generated chunks.

I am talking about a mining rule, which, by the way, should already be implemented for tunneling since it claims:

"Additionally, since it avoids breaking any blocks touching a liquid, it won't break the bottom of a gravel stack below a lava lake (anymore)."

So, it would be nice if this could be enforced for tunneling/area mining as well: Don't break any blocks that are touching lava (and not just at the bottom of the lava.

Basically, it would be nice if Baritone could catch up to OQ Minebot. Here is a vid of how they deal with lava:

https://www.youtube.com/watch?v=Z0VB4PElvRY

They leave it undisturbed.

Even better way would be for the bot to be able to remove lava sources with throwaway blocks.

**Comment by dominikaaaa** [comment]
Avoiding breaking blocks that are touching lava is vastly different and unclear from this issues main contents and title, you should update that.
